app/modules/users/components/profile/profile-details.section.tsx

```diff
@@ -46,6 +53,52 @@ export const ProfileDetailsSection: FC<{
       )}
       render={({ isEdit, setIsEdit }) => (
         <div className="flex flex-col gap-8">
+          <div className="flex items-start gap-4">
+            {nameLabel}
+            {isEdit ? (
+              <Form
+                onSubmit={() => setIsEdit(false)}
+                className="flex items-start gap-4 grow"
+                method="put"
+              >
+                <input type="hidden" name="fields" value="firstName" />
+                <input type="hidden" name="firstName" value={firstName ?? ''} />
+
+                <TextField
+                  label="First Name"
+                  size="small"
+                  variant="outlined"
+                  defaultValue={firstName}
+                  onChange={({ target }) => setFirstName(target.value)}
+                />
+              </Form>
+            ) : (
+              firstName
```

Comment:
Remix will refetch all data after any form resubmit, so you can just use `user.profile?.firstName` instead

There will be a small lag but it's ok to ignore it at this point

p.s.: this will also lead to the fact that you can remove whole state

``` tsx
const [firstName, setFirstName] = useState(user.profile?.firstName);
```

```suggestion
              user.profile?.firstName
```